packages/report-server/src/reportBuilder/output/functions/matrix/matrix.ts

```diff
@@ -16,6 +16,7 @@ const paramsValidator = yup.object().shape(
         ? yup.array().of(yup.string().required())
         : yup.mixed<'*'>().oneOf(['*'], "columns must be either '*' or an array"),
     ),
+    excludeFields: yup.array().of(yup.string().required()),
```

Comment:
So I couldn't think of a clean way to have the output step determine the start/end dates for the data without including the `period` column in the report and treating it as a magic field that the output step reads. A bit hacky, but also a bit pragmatic I feel?

The real downside to me is that if we wish to exclude the `period` column from the report output then we have to manually exclude it here... Tedious and unexpected.

I might try and brainstorm some more to see if there's a cleaner pattern we could use here, but would love if the reviewer had some thoughts?

Reply:
Yeah I can't think of a way without a radically different architecture. Let's walk and talk this though in case we come up with something by bouncing off each other!

Reply:
Sweet, yes lets!